File: src/main/java/org/bakkes/game/view/APositionedView.java
package org.bakkes.game.view;

import org.newdawn.slick.geom.Vector2f;

public abstract class APositionedView extends AView implements IPositionable {

	private final Vector2f position = new Vector2f();
	@Override
	public Vector2f getPosition() {
		return position;
	}
}

File: src/main/scala/org/bakkes/game/view/APositionedView.scala
package org.bakkes.game.view;

import org.bakkes.game.model.TPosition;
import org.newdawn.slick.geom.Vector2f;

abstract class APositionedView() extends AView with TPosition {
	private val position:Vector2f  = new Vector2f()
	override def getPosition(): Vector2f = position
}
